### PR TITLE
Inject loader bootstrap at the root

### PR DESCRIPTION
**Pull request by grouma**
Elements of the Require Loader bootstrap are expected to be globally available, e.g. the [customModulePaths](https://github.com/dart-lang/webdev/blob/master/dwds/lib/src/loaders/require.dart#L225). Previously we were injecting the bootstrap within the main module.

This change cleans up the injection logic so that the bootstrap is injected globally at the end. The injected client must occur inside the root module so I've added documentation to make that clear. I've also removed the logic which changes the hot restart handler as that is no longer used.

### CODE REVIEW

**Comment by grouma** [comment]
This change is blocked on https://github.com/dart-lang/webdev/pull/906

File: dwds/lib/src/handlers/dev_handler.dart [hunk 1]
```diff
@@ -125,9 +125,9 @@ class DevHandler {
       // before events are received.
       unawaited(Future.microtask(() => tabConnection.runtime.enable()));
       // There is no way to calculate the number of existing execution contexts
-      // so we wait for a short while to recieve a context.
+      // so we wait for a short while to receive a context.
       while (await contextQueue.hasNext
-          .timeout(const Duration(milliseconds: 50), onTimeout: () => false)) {
+          .timeout(const Duration(milliseconds: 500), onTimeout: () => false)) {
```
Comment (grouma) [inline dwds/lib/src/handlers/dev_handler.dart]: The injected client is now loaded much earlier so the execution context event takes slightly longer to retrieve.

Comment (jakemac53) [reply dwds/lib/src/handlers/dev_handler.dart]: Do we have an issue filed somewhere about this? Especially when we are running larger apps etc if this scales based on app size or anything its likely to cause problems.

As we have seen with this PR its also likely that other changes in the codebase can arbitrarily break this or make it flaky.

Comment (grouma) [reply dwds/lib/src/handlers/dev_handler.dart]: I do believe we have an issue filed for this. Nonetheless I reverted the change. I have a better handle of what is required. We do have assumptions in `pacakge:dwds` that when the client connects we are in a "ready" state. I've added documentation to make that clear.

**Pull request merged**